### PR TITLE
feat: support preserve specific unused hooks

### LINKED ISSUE #49: Do not remove git hooks
Hi. I use some custom commit-msg and prepare-commit-msg git hooks that grab data from a Jira system. And I use "simple-git-hooks" to run "lint-staged" on "pre-commit". The latest release of "simple-git-hooks" started removing "not-used git hooks" which breaks manually created git hooks. Could you please revert this change, or make it configurable maybe?

### PR DESCRIPTION
**Pull request by JounQin**
fix #49

### CODE REVIEW

File: simple-git-hooks.js [hunk 1]
```diff
@@ -296,14 +300,18 @@ function _getConfigFromFile(projectRootPath, fileName) {
...
- * Validates the config, checks that every git hook is named correctly
+ * Validates the config, checks that every git hook or option is named correctly
  * @return {boolean}
- * @param {{string: string}} hooks
+ * @param {{string: string}} config
  */
-function _validateHooks(hooks) {
+function _validateHooks(config) {
+
+    for (let hookOrOption in config) {
```
Comment (JounQin) [inline simple-git-hooks.js:309]: Should we use `Object.entries` or `Object.keys` here?

Comment (toplenboren) [reply simple-git-hooks.js:309]: I think this syntax looks cleaner, then one with `Object.keys`

File: simple-git-hooks.js [hunk 2]
```diff
@@ -296,14 +300,18 @@ function _getConfigFromFile(projectRootPath, fileName) {
...
+        if (!Object.prototype.hasOwnProperty.call(config, hookOrOption)) {
```
Comment (toplenboren) [inline simple-git-hooks.js:310]: I think we do not need to `continue` in this cycle:

We simply iterate over every entry (hookOrOption) in config, 
If enrty is bad, we raise an error (return false),

File: simple-git-hooks.test.js [hunk 3]
```diff
@@ -172,7 +173,7 @@ test('creates git hooks if configuration is correct from package.json', () => {
 test('fails to create git hooks if configuration contains bad git hooks', () => {
     createGitHooksFolder(projectWithIncorrectConfigurationInPackageJson)
 
-    expect(() => spc.setHooksFromConfig(projectWithIncorrectConfigurationInPackageJson)).toThrow('[ERROR] Config was not in correct format. Please check git hooks name')
+    expect(() => spc.setHooksFromConfig(projectWithIncorrectConfigurationInPackageJson)).toThrow('[ERROR] Config was not in correct format. Please check git hooks or options name')
```
Comment (toplenboren) [inline simple-git-hooks.test.js:176]: 👍

File: simple-git-hooks.js (hunk 1 above)
Comment (JounQin) [reply simple-git-hooks.js:309]: OK.

File: simple-git-hooks.js (hunk 2 above)
Comment (JounQin) [reply simple-git-hooks.js]: @toplenboren Done!

**Comment by JounQin** [comment]
friendly ping @toplenboren

**Review by toplenboren — APPROVED** [review]

**Comment by toplenboren** [comment]
Gonna merge and release a new version tomorrow, thank you!

**Pull request merged**